runtime/src/iree/hal/utils/stream_tracing.c

```diff
@@ -232,6 +232,63 @@ void iree_hal_stream_tracing_context_free(
   IREE_TRACE_ZONE_END(z0);
 }
 
+void iree_hal_stream_tracing_context_collect_list_internal(
```

Comment:
Always put static on any function local to a compilation unit (both to help the compiler know that and for readability):
```suggestion
static void iree_hal_stream_tracing_context_collect_list_internal(
```